### PR TITLE
Fix Kafka::Consumer#finalize argument error

### PR DESCRIPTION
**Pull request by agrare**
Fix a `Nil` pointer argument error in `Kafka::Consumer#finalize` caused by referencing `@topics` in conditional and as an argument without a local var.

```
$ crystal spec
Showing last frame. Use --error-trace for full trace.

In src/kafka/consumer.cr:77:46

77 | LibKafkaC.topic_partition_list_destroy(@topics) if @topics
                                             ^------
Error: argument 'tplist' of 'LibKafkaC#topic_partition_list_destroy' must be Pointer(Void), not (Pointer(Void) | Nil)
```

### CODE REVIEW

File: spec/kafka_spec.cr [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
 describe Kafka do
-  # TODO: add tests
-  # can only really do integration tests, as it requires connecting to working kafka setup
-
+  describe Kafka::Consumer do
+    context "#initialize" do
+      it "returns a Kafka::Consumer" do
+        config = Kafka::Config.new
+        kafka = Kafka::Consumer.new(config)
+        kafka.should be_a(Kafka::Consumer)
```
Comment (agrare) [inline spec/kafka_spec.cr:10]: NOTE: This really isn't much of a test, just wanted something which would show the failure

File: src/kafka/consumer.cr [hunk 2]
```diff
@@ -74,7 +74,9 @@ module Kafka
     # :nodoc:
     def finalize
       LibC.free(@p_err_str)
-      LibKafkaC.topic_partition_list_destroy(@topics) if @topics
+      if topics = @topics
+        LibKafkaC.topic_partition_list_destroy(topics)
```
Comment (agrare) [inline src/kafka/consumer.cr]: I see you use the `@topics.try {}` style in the producer, I'll update to use that

**Comment by agrare** [comment]
@philipp-classen would you mind taking a look?

**Comment by philipp-classen** [comment]
@agrare Apology, I did not see your pull request. Thanks for it, I'll have a look.

**Comment by philipp-classen** [comment]
@agrare Your idea is sound. I tried to simplify the original code a bit further. Please have a look at https://github.com/philipp-classen/kafka.cr/pull/2

It still passes your tests. Please have a look that I did not introduce some regressions. If not I would merge that updated version.

Thanks for spotting the problem!

**Comment by agrare** [comment]
:+1: your change works for me thanks so much @philipp-classen 
Closing in favor of #2

**Pull request closed without merging**